DataSet strict mode, UMS Namespaces, Policy Filters, Delete project (#28)
jexia/docs: docs/auth/README.md

</template>
</CodeSwitcher>

::: warning

> Keep in mind at this point in time only JS SDK is automatically refresh access token. If you use REST API or Python SDK you need to handle this situation by your own. Token live time 2h 

Reviewer: ```suggestion
Keep in mind at this point in time only the JS SDK automatically refreshes the access token. If you use the REST API or the Python SDK you need to handle this situation yourself. Each token lasts 2 hours. 
```

:::

## Project Users
Another way to authenticate within Jexia is to use **Project Users**. Usually, you need this when you want to provide more rights to specific users. Let's say **Update** and **Delete** actions for your blog can be done by owners only. Firstly, you need to create a user under the **Project Users** section.

![UMS Users](./ums-2.png)

> After this, you need go to the **Policies** section and create a new policy, ensuring that **Subject** is selected for some specific **Namespace** or **User account** and **Resource** are selected to specific dataset, fileset or AllUsers(for user table) what you want these users be able to run actions. Finally, select all **Actions** you wanted these users to perform.

Reviewer: ```suggestion
After this, you need go to the **Policies** section and create a new policy, ensuring that **Subject** is selected for a specific **Namespace** or **User account** and the **Resource** is selected to the specific dataset, fileset or the 'AllUsers' option (for user table) if you want these users to be able to run the actions. Finally, select all **Actions** you wanted these users to perform.
```

![Policy](./policy.png)

> This will allow any user from namespace `registered` or user with email `admin@x.com` to have Update & Delete operations for all own records and Read operation for all records which has schema field `confirmed` equal to `true`

Reviewer: ```suggestion
This will allow any user from the namespace `registered` or the user with the email `admin@x.com` to have **Update & Delete** operations for all their records and the **Read** operation for all the records which have a schema field `confirmed` equal to `true`.
```

</template>
</CodeSwitcher>

::: warning

> Keep in mind at this point in time only JS SDK is automatically refresh access token. If you use REST API or Python SDK you need to handle this situation by your own. Token live time 2h 

Reviewer: ```suggestion
Keep in mind at this point in time only the JS SDK automatically refreshes the access token. If you use the REST API or the Python SDK you need to handle this situation yourself. Each token lasts 2 hours. 
```

> Keep in mind at this point in time only JS SDK is automatically refresh access token. If you use REST API or Python SDK you need to handle this situation by your own. Token live time 2h 

Reviewer 2: Maybe formulate this a bit different to: 
Keep in mind at this point in time only JS SDK automatically refreshes the access token. If you use REST API or Python SDK you need to handle this yourself. The token does have a duration length of 2h

Author: it is by purpose to have 2 times. I think Bailey adjusted this part as well

:::

## Policies
As was mentioned above to authorize access to your data you need to have a **policy** created for your API keys and/or Project Users. 
Currently, you cannot manipulate a **policy** via the API. All admin actions are only available via Jexia's administration dashboard.

<iframe width="700" height="394" src="https://www.youtube.com/embed/i4dKznoXry0" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

_* video shows previous version of policy screen_

There are four main areas within a **Policy**:

> 1. **Subject** (who has access) - can be an **API key**, **Namespace** (grouped users under the same name), **User account**.

Reviewer: ```suggestion
1. **Subject** (who has access) - can be an **API key**, **Namespace** (grouped users under the same name) or a **User account**.
```

2. **Resources** (access to what) - can be any dataset, fileset or channel. Selecting **All Users** means you allow  operations for all Project Users. 
3. **Actions** (what CRUD actions can be performed) - here you can specify which actions are allowed to be performed. These can be: Create, Read, Update and Delete. For Channels you have specific actions: Subscribe (read) and Publish (write). 

> 4. **Filters** - for actions you can use additional filtering options to specify access of current namespace, api or user. It is useful when you want to segrigate data access for different instances of your business(offices, e-shops, etc).

Reviewer: ```suggestion
4. **Filters** - for actions you can use additional filtering options to specify access of current namespace, API or user. It is useful when you want to segrigate data access for different instances of your business, such as offices and e-shops etc.
```

> We have predefined filter - Owner which says to the platform to filter out only records where user is an owner. It can combine with additional filters to specify even more. All filters joind with **AND** logical operator. Changes for policy applies immediately.

Reviewer: ```suggestion
We have a predefined filter - **Owner** which filters and shows only records where the user is an owner. It can be combined with additional filters to specify even more. All filters join with the **AND** logical operator. Changes for a policy are applied immediately.
```

> We have predefined filter - Owner which says to the platform to filter out only records where user is an owner. It can combine with additional filters to specify even more. All filters joind with **AND** logical operator. Changes for policy applies immediately.

Reviewer 2: Maybe change: "We have predefined filter" to "We have a predefined filter" and: "it can combine with additional filters" to "It can be combined with additional filters".

Author: Reviewed by Bailey as well, see the commit history

![Policy](./policy.png)

> If namespace or particular user is part of multiple policies, access will be joined via **OR** operations. 

Reviewer: ```suggestion
If a namespace or particular user is part of multiple policies, access will be joined via the **OR** operator. 
```
